content/topics/code-comments/_index.md

```diff
@@ -0,0 +1,39 @@
+---
+content_type: topic
+ready: true
+title: Code comments
+---
+
+## Understanding Effective Comments in Coding
+
+## Introduction
+
+In programming, comments play a crucial role in code documentation. They provide explanations, clarifications, and additional context to make the code more understandable for developers. However, not all comments are created equal. It is important to distinguish between good comments and bad comments to ensure effective code communication. Let's explore what makes a comment good or bad.
+
+### Characteristics of a Good Comment
+
+1. Purposeful Explanation: A good comment serves the purpose of explaining complex logic, algorithms, or business rules. It provides insights into the code's functionality and helps other developers understand the intentions behind certain decisions.
+
+2. Clarity and Readability: A good comment is clear, concise, and easy to read. It uses proper grammar, punctuation, and formatting conventions. Well-structured comments make it easier for developers to follow the code and comprehend its meaning.
+
+3. Relevant Information: A good comment provides relevant information that is not immediately apparent from the code itself. It adds value by offering insights, reasoning, or describing the intention behind a particular approach or solution.
+
+4. Updates with Code Changes: A good comment is maintained and updated alongside the code. If the code undergoes modifications, the corresponding comments should be reviewed and revised to ensure they accurately reflect the updated logic.
+
+5. Avoidance of Redundancy: A good comment avoids repeating what is already evident from the code. It focuses on explaining aspects that may not be immediately obvious, such as complex algorithms or non-standard approaches.
+
+### Characteristics of a Bad Comment
+
+1. Irrelevance: A bad comment lacks relevance to the code or provides information that is obvious from reading the code itself. Irrelevant comments clutter the codebase and can confuse or mislead developers.
+
+2. Outdated or Inaccurate Information: A bad comment contains outdated or incorrect information. If a comment no longer aligns with the current code logic or behavior, it becomes misleading and should be removed or updated accordingly.
+
+3. Vague or Ambiguous Descriptions: A bad comment fails to provide clear explanations or uses ambiguous language. Unclear comments can cause confusion and hinder code comprehension instead of enhancing it.
+
+4. Excessive or Unnecessary Comments: A bad comment includes excessive or unnecessary explanations that only add noise to the code. Comments should be used sparingly, focusing on areas that genuinely require clarification or insight.
+
+5. Sarcasm, Insults, or Personal Opinions: A bad comment contains sarcasm, insults, or personal opinions that are unprofessional and create a negative work environment. Comments should maintain a respectful and collaborative tone.
+
```

Comment:
Maybe add a section on common problems. Eg new programmers often:
- just paraphrase code in their comments. Eg: `sum = 0 // initialise the sum variable to 0`
- leave out comments completely because people keep telling them to stop making useless comments

It might also be useful to talk about who the comments are for. Comments are for coders - if someone who already knows how to code reads a comment then it should make their life easier somehow.

Reply:
It would also be good to mention something about self-documenting code.  

Good code makes it's intentions clear. It does this by:
- using clear algorithms
- using well chosen names
- comments as needed

A common problem is something like this:
```
// calculate the temperature in fereignheit given the temperature in celcius
functrion calculate(temp){
     return .... 
}
```
If the code was just better then the comment wouldn't be needed at all.

Reply:
Generally, it's good to include examples instead of just explanations. It makes it easier for people to grasp what you are talking about.